views/on-ramp/QuotePreview.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div
+    class="cursor-pointer rounded-xl border border-gray-200 bg-gray-100 hover:bg-gray-200/50 dark:border-neutral-800/70 dark:bg-neutral-900 dark:hover:bg-neutral-800/80"
+    @click="runQuote"
+  >
+    <div class="flex gap-2 p-3">
+      <div class="basis-2/3">
+        <div>
+          <div :title="balance[1]" class="font-bold">{{ balance[0] }} {{ quote.receive.token.symbol }}</div>
+          <div class="text-sm text-gray-600 dark:text-gray-300">
+            Fee: {{ formatFiat(quote.pay.totalFeeUsd, quote.pay.currency) }}
+          </div>
+          <button
+            type="button"
+            class="p-0.5 text-sm text-neutral-500 underline hover:text-neutral-600 dark:text-neutral-500 dark:hover:text-neutral-400"
+            @click="toggleDetails"
+          >
+            Details
+          </button>
+        </div>
+      </div>
+      <div class="flex basis-1/3 items-center justify-end">
+        <!-- <div class="inline-block p-2">
+          <img :src="quote.provider.iconUrl" class="h-8 w-8" />
+        </div> -->
+        <div class="inline-block">
+          <div>{{ quote.provider.name }}</div>
+          <div class="text-sm text-gray-600 dark:text-gray-300">{{ providerType }}</div>
+        </div>
+      </div>
+    </div>
+    <Transition v-bind="TransitionOpacity(250, 150)">
+      <div v-if="toggleOpen" ref="quotePreview">
+        <StepDetail v-for="(step, index) in quote.steps" :key="index" :step="step" />
+      </div>
+    </Transition>
```

Comment:
```suggestion
<CommonHeightTransition :opened="toggleOpen">
      <StepDetail v-for="(step, index) in quote.steps" :key="index" :step="step" />
    </CommonHeightTransition>
```

It is possible to use `CommonHeightTransition` to animate height via css only.

Reply:
Would this allow to get rid of `middlePanelHeight`?

Reply:
`middlePanelHeight` is needed for transitioning between different component views in that section. Without it, height will not work correctly to expand or shrink based on the content inside when views change.